tools/qa/gen_regression_test.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# HORTON: Helpful Open-source Research TOol for N-fermion systems.
+# Copyright (C) 2011-2016 The HORTON Development Team
+#
+# This file is part of HORTON.
+#
+# HORTON is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# HORTON is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+#
+# --
+"""Script to generate regression tests.
+
+To use, simply pass the name of the example python script.  It will write a testing script in your
+current directory with name test_<orig_script_name>.py, which can be used with nosetests.
+
+There are several limitations.
+1) All of the variable types to be checked must by numpy compatible. You cannot specify a class.
+For things like integrals, access the _array attribute if necessary.
+2) The original script must have variables named starting with "result_". These will be
+checked.
+3) If the variables to be checked need tolerances, they must be named starting with
+"threshold_" and the name must match the corresponding "result_" variable.
+4) The original example script cannot be moved, or the regression tests must be regenerated.
+5) The variables get printed into the unit test. Try to avoid dumping large things like
+two-electron integrals.
+"""
+import sys
+import numpy as np
+
+
+def gen_regression_test():
+    """Writes regression test file. See module docstring."""
+    # Set global configurations
+    np.set_printoptions(threshold=np.inf)
+
+    # Set defaults for path
+    default_threshold = 1e-8
+
+    # Set path to operate on
+    test_path = sys.argv[1]
+
+    # Generate reference values
+    with open(test_path) as fh:
+        exec fh  # pylint: disable=exec-used
```

Comment:
It would be safer to use the additional arguments of the `exec` statement, e.g. as follows:

    variables = {}
    with open(test_path) as fh:
        exec fh, variables

will store all locals and globals of the example script in a dictionary, which avoids potential name collisions with variable names in this script.